master/pkg/schemas/expconf/legacy.go

```diff
@@ -136,6 +142,31 @@ func getEnvironmentVariables(raw map[string]interface{}) (EnvironmentVariablesMa
 	return ev, nil
 }
 
+func getPodSpec(raw map[string]interface{}) (*PodSpec, error) {
+	ps := &PodSpec{}
+
+	envOnly := raw["environment"]
+	if envOnly == nil {
+		return nil, nil
+	}
+
+	evOnly := envOnly.(map[string]interface{})["pod_spec"]
+	if evOnly == nil {
+		return nil, nil
+	}
+
+	rawBytes, err := json.Marshal(evOnly)
+	if err != nil {
+		return ps, errors.Wrap(err, "unable to remarshal environment as json")
```

Comment:
probably should return `nil, error`, since it's a `*PodSpec` and not a `PodSpec` we're returning.